### PR TITLE
MID changes in reference app

### PR DESCRIPTION
**Pull request by AbhiramE151859**

### CODE REVIEW

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 1]
```diff
@@ -93,18 +97,33 @@ public static List<Match> getMultipleMatches() throws UnrecoverableKeyException,
...
+                null,
+                null,
+                null,
+                USA_COUNTRY_CODE,
+                EXAMPLE_ID_TYPE,
+                EXAMPLE_ID_VALUE
         );
     }
-
     public static List<Match> throwsNoMatchFound() throws UnrecoverableKeyException, CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, ApiException {
```
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:117]: can we have throwsNoMatchFoundByMID() method as well?

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 2]
```diff
@@ -137,7 +160,22 @@ public static List<Match> throwsInvalidCountryCodeApiException() throws Unrecove
                 EXAMPLE_POSTAL_CODE,
                 EXAMPLE_CITY,
                 EXAMPLE_STATE_PROVINCE_CODE,
-                "AUS" // this must be USA_COUNTRY_CODE for the current release of the API
+                "AUS",// this must be USA_COUNTRY_CODE for the current release of the API
```
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:163]: change comment to "this must be one of the ISO standard country codes supported by SBCA application"

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 3]
```diff
@@ -137,7 +160,22 @@ public static List<Match> throwsInvalidCountryCodeApiException() throws Unrecove
...
+                null,
+                null
+        );
+    }
+
+    public static List<Match> throwsInvalidIDType() throws UnrecoverableKeyException, CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, ApiException {
```
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:169]: pass some invalid id type for this case like "MID" in place of empty

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 4]
```diff
@@ -137,7 +160,22 @@ public static List<Match> throwsInvalidCountryCodeApiException() throws Unrecove
...
+        return new MatchingApi(getApiClient()).getMatches(
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                "",
+                EXAMPLE_ID_VALUE
```
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:178]: add throwsInvalidIDValue() example as well

File: src/main/resources/small-business-credit-analytics.yaml [hunk 5]
```diff
@@ -167,27 +169,32 @@ components:
               Details: >-
                 The specified location could not be found in Mastercard's
                 merchant metrics database.
+    CountryCodeNotPermittedExample:
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:172]: we can remove this example. I think we are not having this test case either.

File: src/main/resources/small-business-credit-analytics.yaml [hunk 6]
```diff
@@ -309,6 +322,17 @@ components:
           examples:
             UnauthorizedExample:
               $ref: '#/components/examples/UnauthorizedExample'
+    CountryCodeNotPermitted:
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:325]: remove this example

File: src/main/resources/small-business-credit-analytics.yaml [hunk 7]
```diff
@@ -220,12 +221,27 @@ components:
...
+    idTypeParameter:
+      name: id_type
+      in: query
+      description: >-
+        User input for ID bases searching
+      example: MERCHANT_ID
+      schema:
+        type: string
+        enum:
+          - 'MERCHANT_ID'
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:236]: do we need to maintain enum? This validation is taken care by SBCA Matches API.

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 8]
```diff
@@ -75,36 +80,92 @@ public static ApiClient getApiClient() throws UnrecoverableKeyException, Certifi
...
+                null,
+                null,
+                null,
+                null,
+                USA_COUNTRY_CODE,
+                EXAMPLE_ID_TYPE_MERCHANT_ID,
+                SINGLE_MATCH_EXAMPLE_ID_VALUE_MERCHANT_ID_VALUE
+        );
+    }
+    public static List<Match> getMultipleMatchByMID() throws UnrecoverableKeyException, CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, ApiException {
```
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:95]: change this to getMultipleMatch**es**ByMID

File: src/main/resources/small-business-credit-analytics.yaml [hunk 9]
```diff
@@ -196,36 +192,43 @@ components:
...
       in: query
       description: >-
         The state or province abbreviation in the merchant's address. Expected
         conventions will be determined based on country_code - e.g. for USA a
         2-letter abbreviation is expected.
       example: NY
-      required: true
       schema:
         type: string
         pattern: '^[0-9A-Z]+'
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:215]: on state_province_region can you remove this regex validation?

File: src/main/resources/small-business-credit-analytics.yaml [hunk 10]
```diff
@@ -196,36 +192,43 @@ components:
         determined based on country_code - e.g. for USA a 5-digit or 9-digit
         postal code is expected.
       example: 12345
-      required: true
       schema:
         type: string
         pattern: '^[0-9A-Z- ]+'
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:197]: remove regex on postal_code

File: src/main/resources/small-business-credit-analytics.yaml [hunk 11]
```diff
@@ -173,21 +175,15 @@ components:
       in: query
       description: Name of merchant (DBA name).
       example: Fancy Restaurant
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:177]: can we update this example value to single match results?

File: src/main/resources/small-business-credit-analytics.yaml [hunk 12]
```diff
@@ -234,12 +237,9 @@ components:
         follow https://www.iso.org/iso-3166-country-codes.html, but only
         currently-supported countries will be accepted.
       example: USA
-      required: true
       schema:
         type: string
         pattern: '^[A-Z]+'
```
Comment (niduramurali) [inline src/main/resources/small-business-credit-analytics.yaml:242]: remove the regex on country_code. Let customers experience error messages from SBCA Matching API

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java (hunk 8 above)
Comment (niduramurali) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:95]: change this to reflect plural getMultipleMatch**es**ByMID

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java
Comment (srinivasa-t) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:1]: @AbhiramE151859 why do we need to pass null for ID type and ID value when searching by name and address, why do we need have an overloaded method.
FYI @niduramurali

File: src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java [hunk 13]
```diff
@@ -75,36 +81,105 @@ public static ApiClient getApiClient() throws UnrecoverableKeyException, Certifi
...
+                null,
+                null,
+                null,
+                null,
+                null,
+                USA_COUNTRY_CODE,
+                EXAMPLE_ID_TYPE_MERCHANT_ID,
+                SINGLE_MATCH_EXAMPLE_ID_VALUE_MERCHANT_ID_VALUE
+        );
+    }
```
Comment (srinivasa-t) [inline src/main/java/com/mastercard/creditanalytics/api/ApiExamples.java:95]: @AbhiramE151859 why do we need to pass null for name and address fields  when searching by ID, why do we not have have an overloaded method.
FYI @niduramurali

**Review by srinivasa-t — CHANGES_REQUESTED** [review]
@AbhiramE151859 I have reviewed and added comments.

**Pull request merged**